File: 539-minimum-time-difference/539.js
/**
 * 539. Minimum Time Difference
 * https://leetcode.com/problems/minimum-time-difference/
 * Javascript solution
 */

/**
 * @param {string[]} timePoints
 * @return {number}
 */
const findMinDifference = (timePoints) => {
  // convert to minutes & int
  let minArr = [];
  for (i = 0; i < timePoints.length; i++)
    minArr.push(Number(timePoints[i].split(':')[0]) * 60 + Number(timePoints[i].split(':')[1]))

  // sort min -> max
  minArr = minArr.sort((f, s) => f - s);

  // map minuteArr (minArr) to differences between adjacent minutes, then return minimum difference
  minArr = minArr.map((val, i, arr) => {
    if (arr[i + 1] !== undefined) {
      return arr[i + 1] - val
    }
    else {
      return 1440 - val + arr[0]
    }
  });

  return Math.min.apply(null, minArr)
}

File: 539-minimum-time-difference/539.py
# 539. Minimum Time Difference
# https://leetcode.com/problems/minimum-time-difference/
# Python solution

class Solution:
    def findMinDifference(self, timePoints: List[str]) -> int:
        # convert to minutes & int
        minArr = [int(point.split(':')[0])*60 + int(point.split(':')[1]) for point in timePoints]

        # sort min -> max
        minArr.sort()

        # compare adjacent values in array to find minimum; return minimum
        self.min = minArr[-1]
        for i in range(len(minArr)):
            diff = 0
            try:
                diff = minArr[i+1] - minArr[i]
            except IndexError:
                diff = 1440 - minArr[i] + minArr[0]
            self.min = diff if diff < self.min else self.min
        return self.min
